src/store/ShareStore.js

```diff
@@ -39,14 +39,10 @@ export default class ShareStore {
     @observable urlGenerated = false;
     @observable shortUrlFailed = false;
     @observable isLoadingPNG = false;
-
     @observable shareLink = '';
 
-    fixedEncodeURIComponent(str) {
-        return encodeURIComponent(str).replace(/[!'()*]/g, c => `%${c.charCodeAt(0).toString(16)}`);
-    }
-    refereshShareLink = () => {
-        const self = this;
+
+    @action refereshShareLink = () => {
```

Comment:
shouldn't this be `@action.bound refreshShareLink(str) {`?

also do fix the typo. Aside that this looks good to me